codecoverage.py

```diff
@@ -208,24 +208,22 @@ def main():
     parser.add_argument('branch', action='store', nargs='?', default=default_branch, help='Branch on which jobs ran')
     parser.add_argument('commit', action='store', nargs='?', default=default_commit, help='Commit hash for push')
     parser.add_argument('--grcov', action='store', nargs='?', help='Path to grcov')
-    parser.add_argument('--no-download', action='store_true', help='Use already downloaded coverage files')
-    parser.add_argument('--no-grcov', action='store_true', help='Use already generated grcov output (implies --no-download)')
+    parser.add_argument('--with-artifacts', action='store', nargs='?', help='Path to already downloaded coverage files')
+    parser.add_argument('--no-grcov', action='store_true', help='Use already generated grcov output (implies --with-artifacts PATH_TO_ARTIFACTS)')
     parser.add_argument('--suite', action='store', nargs='+', help='List of test suites to include (by default they are all included). E.g. \'mochitest\', \'mochitest-chrome\', \'gtest\', etc.')
     parser.add_argument('--ignore', action='store', nargs='+', help='List of test suites to ignore (by default \'talos\' and \'awsy\'). E.g. \'mochitest\', \'mochitest-chrome\', \'gtest\', etc.')
     args = parser.parse_args()
 
-    if args.no_grcov:
-        args.no_download = True
-
-    if (args.branch is None) != (args.commit is None) and not args.no_download:
-        parser.print_help()
-        return
+    if not args.with_artifacts:
+        if (args.branch is None) != (args.commit is None) or args.no_grcov:
```

Comment:
If there's no `with_artifacts` option, you should still be able to use `no_grcov`, so you can remove `or args.no_grcov`